esmvaltool/diag_scripts/landcover/albedolandcover.py

```diff
@@ -0,0 +1,338 @@
+import cartopy.feature as cfeature
+import iris
+import matplotlib.pyplot as plt
+import numpy as np
+# specific imports for this diagnostic
+from sklearn import linear_model
+
+from esmvaltool.diag_scripts.shared import group_metadata, run_diagnostic
+from esmvaltool.diag_scripts.shared._base import get_plot_filename
+
+
+# This part sends debug statements to stdout
+logger = logging.getLogger(os.path.basename(__file__))
+logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
+
+
+# This suppresses a warning. Check if this is legitimate.
+iris.FUTURE.netcdf_promote = True
+
+warnings.filterwarnings("ignore", category=FutureWarning)
+
+
+def _add_masks_albedolandcover(model_data, this_models_xxfracs, dia_cfg, cfg):
+
+    total_frac = sum([model_data[key] for key in this_models_xxfracs])
+
+    # Mask out regions where total_frac is too low
+    fracmask = (total_frac.data.data < dia_cfg['threshold_sumpred'])
+
+    # Start masking operations. Remember that a True means masked out.
+    basemask = model_data['snc'].data.mask
+
+    # Mask out regions where there is little snow
+    snowmask = model_data['snc'].data.data < dia_cfg['thres_fsnow']
+    snowfreemask = ~snowmask
+
+    # Update the masks
+    snowmask |= basemask
+    snowmask |= fracmask
+    snowfreemask |= basemask
+    snowfreemask |= fracmask
+
+    # Plotting intermezzo for the masks
+    masksavedir = os.path.join(cfg['plot_dir'], 'masks/')
+    if not os.path.exists(masksavedir):
+        os.mkdir(masksavedir)
+    else:
+        print("Dir exists, be aware of overwriting")
+
+    template_time = model_data['snc'].coord('time')
+    month_string = template_time.units.num2date(
+        template_time.points)[0].strftime('%b')
+    masksavename = '{0}-{1}'.format(
+        month_string, model_data['snc'].attributes['model_id'])
+    plt.imshow(total_frac.data[::-1])
+    plt.savefig(masksavedir + masksavename + 'total_frac.'
+                + cfg['output_file_type'])
+    plt.imshow(fracmask[::-1])
+    plt.savefig(masksavedir + masksavename + 'fracmask.'
+                + cfg['output_file_type'])
+    plt.imshow(snowmask[::-1])
+    plt.title('snowmask')
+    plt.savefig(masksavedir + masksavename + 'snowmask.'
+                + cfg['output_file_type'])
+    plt.imshow(snowfreemask[::-1])
+    plt.title('snowfreemask')
+    plt.savefig(masksavedir + masksavename + 'snowfreemask.'
+                + cfg['output_file_type'])
+
+    # TODO also add the snowmask
+    mymask = copy.deepcopy(snowfreemask)
+
+    for varkey in model_data:
+        model_data[varkey].data.mask = mymask
+
+    return model_data
+
+def _get_empty_lc_array(model_data):
+    alb_lc1 = copy.deepcopy(model_data['alb'].data)
+    alb_lc1[...] = np.nan  # Set all data to NaN
+    alb_lc2 = copy.deepcopy(alb_lc1)
+    alb_lc3 = copy.deepcopy(alb_lc1)
+    alb_lc = np.stack((alb_lc1, alb_lc2, alb_lc3))
+    return alb_lc
+
```

Comment:
can you construct an empty array on the fly pls, without copying data into memory - it is more efficient to build your array as `np.empty` or `np.zeros(shape, bool=True)` than copying data